integration_tests/kpi.feature

```diff
@@ -56,12 +56,18 @@ Feature: KPI integration tests
      Then self_sufficiency of {'Grid': 1.0, 'Community': 0.666, 'House 1': 0.0, 'House 2': 0.5, 'House 3': 1.0} are correctly reported
      And self_consumption of {'Grid': 1.0, 'Community': 0.8, 'House 1': None, 'House 2': 1.0, 'House 3': 0.5} are correctly reported
 
+  """
+  Feedback required:
+  Earlied PAB was contiuously matching and announcing trade till it has been exhausted, but now it simply forwards the
+  recommended pair and rematch the derivative bids/offers in the next tick, due to which there are chances of unmatched
+  loads at the last tick leading to reduction in SS/SC
+  """
```

Comment:
I think I need to understand this before I can approve. Can you maybe elaborate a bit more? Maybe with an example?

Reply:
Earlier PAB algo was simultaneously creating a match and entering into trade until all bids and offers have exhausted

```
    def match_offers_bids(self):
        while len(self._perform_pay_as_bid_matching()) > 0:
            for bid, offer in self._perform_pay_as_bid_matching():
                selected_energy = bid.energy if bid.energy < offer.energy else offer.energy
                original_bid_rate = bid.original_bid_price / bid.energy
                matched_rate = bid.energy_rate

                trade_bid_info = TradeBidOfferInfo(
                    original_bid_rate=original_bid_rate,
                    propagated_bid_rate=bid.price/bid.energy,
                    original_offer_rate=offer.original_offer_price/offer.energy,
                    propagated_offer_rate=offer.price/offer.energy,
                    trade_rate=original_bid_rate)

                self.accept_bid_offer_pair(bid, offer, matched_rate,
                                           trade_bid_info, selected_energy)
```
Now it only sends back the recommended pair, and the derived offers/bids from residual trade is recommended at the next tick

Reply:
I see. I wonder why the other KPI tests dont fail then. Why only this one where there are multiple levels?

Reply:
What I understood of the reason is its the one without pv/power_producing device in one area, so the residual offers from other area also need time to propagate too

Reply:
Sorry, do not understand `the one without pv/power_producing device in one area` It has indeed PVs in all Houses. Can you please investigate more, because I dont like to shrug it of that easily. Backward compatibility should either be given or at least the reason for the change in behaviour should be known.

Reply:
Please give `kpi.d3asim_2103_nopv1_2_sided_pab` being used in this scenario and then re-read my last comment.

Reply:
Ah yes, sorry, I had a look at the wrong setup file. OK, But lets look at the numbers here. I may understand why the self-sufficiency of the House2 is reduced (sending PV energy to somewhere else). However, by definition, the self-consumption of the Grid should be 1 in any case, right? There is no external entity that provides energy. 
Or am I missing something here?

Reply:
@hannesdiedrich Please give this commit a look https://github.com/gridsingularity/d3a/pull/1121/commits/d7ddb9f38cab1706fd2e98791f112b42378d1291

Reply:
Took me a while to understand but, great! Nice catch. This was falsely added when we added the seller_id to the bids and trades, right?
Why wasn't this an issue before i wonder? Do you have an explanation?
This proves that the integrationtests we have at the moment should not be changed (apart from something major was changed) . If an integrationtest fails, there must be something wrong.

Reply:
> Took me a while to understand but, great! Nice catch. This was falsely added when we added the seller_id to the bids and trades, right?

Yes, from commit history, its been like that since I introduced `seller_id`

> Why wasn't this an issue before i wonder? Do you have an explanation?

`seller_id/buyer_id/buyer_origin_id` all are only relevant for ´kpi´, only explanation I can have is that it appeared now due to `self-consumption` ended-up being less than 1 for root area